concent_api/concent_api/tests/test_middleware.py

```diff
@@ -81,3 +106,229 @@ def test_golem_messages_version_middleware_should_attach_http_header_to_response
+        self
+    ):
+        with mock.patch('core.views.handle_messages_from_database', side_effect=CustomException(), autospec=True):
+            response = self.client.post(
+                reverse('core:receive'),
+                data=self._create_client_auth_message(PROVIDER_PRIVATE_KEY, PROVIDER_PUBLIC_KEY),
+                content_type='application/octet-stream',
+                headers={
+                    "Accept": "text/html"
+                }
+            )
+
+            self.assertEqual(response.status_code, 500)
+            self.assertEqual(response.reason_phrase, 'Internal Server Error')
+            self.assertEqual(response._headers['content-type'][1], 'text/html')
+            self.assertIn(b"STACK TRACE", response.content)
+
+    def _assert_proper_internal_server_error_received(self, response, loaded_json, error_message, error_code):
+        self.assertEqual(response.status_code, 500)
+        self.assertIn('error_message', loaded_json)
+        self.assertEqual(loaded_json['error_message'], error_message)
+        self.assertEqual(loaded_json['error_code'], error_code)
+
+
+class TestShouldReturnHtml(unittest.TestCase):
+    def test_that_with_no_information_in_request_html_is_not_used(self):
+        request = mock.Mock()
+        request.META = {}
+
+        is_html_used = should_use_html(request)
+
+        self.assertFalse(is_html_used)
+
+    def test_that_with_empty_headers_in_request_html_is_not_used(self):
+        request = mock.Mock()
+        request.META = {
+            'headers': {}
+        }
+
+        is_html_used = should_use_html(request)
+
+        self.assertFalse(is_html_used)
+
+    def test_that_with_no_acceptance_of_html_in_request_html_is_not_used(self):
+        request = mock.Mock()
+        request.META = {
+            'headers': {
+                'Accept': ["sth/else/than/html"]
+            }
+        }
+
+        is_html_used = should_use_html(request)
+
+        self.assertFalse(is_html_used)
+
+    def test_that_with_acceptance_of_html_in_request_html_is_used(self):
+        request = mock.Mock()
+        request.META = {
+            'headers': {
+                'Accept': ["text/html"]
+            }
+        }
+
+        is_html_used = should_use_html(request)
+
+        self.assertTrue(is_html_used)
+
+    def test_that_with_acceptance_of_html_and_json_with_same_weights_in_request_html_is_not_used(self):
+        request = mock.Mock()
+        request.META = {
+            'headers': {
+                'Accept': ["text/html", "application/json"]
+            }
+        }
+
+        is_html_used = should_use_html(request)
+
+        self.assertFalse(is_html_used)
+
+    def test_that_with_acceptance_of_html_with_greater_weight_than_json_in_request_html_is_used(self):
+        request = mock.Mock()
+        request.META = {
+            'headers': {
+                'Accept': ["text/html;q=0.9", "application/json;q=0.8"]
```

Comment:
This test must be broken. The implementation has a bug because it checks for `:q=`, not `;q=`. If the test passes then it's too weak.

Reply:
there was a bug in the tests. Fixed ;)